deploy-infra/deploy-caa.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+set -euo pipefail
+# Check if the DEBUG env var is set to true
+if [ "${DEBUG:-false}" = "true" ]; then
+  set -x
+fi
+
+# Compulsory env vars
+: "${AZURE_RESOURCE_GROUP}:?"
+: "${SSH_KEY}:?"
+
+AZURE_REGION=${AZURE_REGION:-eastus}
+CLUSTER_NAME="${CLUSTER_NAME:-caaEncryptedImages}"
+
+AZURE_SUBSCRIPTION_ID=$(az account show --query id --output tsv)
+AKS_RG="${AZURE_RESOURCE_GROUP}-aks"
+registry="quay.io/confidential-containers"
+AZURE_IMAGE_ID="/CommunityGalleries/cocopodvm-d0e4f35f-5530-4b9c-8596-112487cdea85/Images/podvm_image0/Versions/$(date -v -1d "+%Y.%m.%d" 2>/dev/null || date -d "yesterday" "+%Y.%m.%d")"
+AZURE_WORKLOAD_IDENTITY_NAME="caa-identity"
+
+export USER_ASSIGNED_CLIENT_ID="$(az identity show \
+  --resource-group "${AZURE_RESOURCE_GROUP}" \
+  --name "${AZURE_WORKLOAD_IDENTITY_NAME}" \
+  --query 'clientId' \
+  -otsv)"
+
+export AZURE_VNET_NAME=$(az network vnet list \
+  --resource-group "${AKS_RG}" \
+  --query "[0].name" \
+  --output tsv)
+
+export AZURE_SUBNET_ID=$(az network vnet subnet list \
+  --resource-group "${AKS_RG}" \
+  --vnet-name "${AZURE_VNET_NAME}" \
+  --query "[0].id" \
+  --output tsv)
+
+export CLUSTER_SPECIFIC_DNS_ZONE=$(az aks show \
+  --resource-group "${AZURE_RESOURCE_GROUP}" \
+  --name "${CLUSTER_NAME}" \
+  --query addonProfiles.httpApplicationRouting.config.HTTPApplicationRoutingZoneName -otsv)
+
+# Ensure we always clone the kbs code base in the kbs sub directory.
+cd $(dirname "$0")
+
+# Pull the CAA code
+if [ ! -d cloud-api-adaptor ]; then
+  git clone https://github.com/confidential-containers/cloud-api-adaptor
```

Comment:
I think here also we can pin a specific commit of CAA. Since 0.8.0 is close, should we wait for the release and then pin to the 0.8.0 release ?

Reply:
Right now pinning the latest main: b2b64c3c2de12dc25aa33e503d2036d2aca33204. I will test this today again to see if there are any issues.